### PR TITLE
updates to pytest coverage

### PR DESCRIPTION
**Pull request by ashish-jabble**
- [x] pytest-cov pip dependency version updated in requirements
- [x] .coveragerc - settings configuration file added. omit has contains unused python dead code
- [x] .pytest.ini - pytest configuration added (this has very minimal at the moment and will add more options by the time)
- [x] python/.coveragerc removed and got added under `tests/` dir

For all python unit tests Coverage report CI shows: **73%** but with this branch actual coverage report is **56%**

### CODE REVIEW

**Comment by Mohit04tomar** [comment]
is there any test job run on latest commit this pull request?

I ran this nightly test:
http://jenkins.dianomic.com:8080/view/Fledge%20Nightly%20Jobs/job/fledge_nightly_build_ub2004/1248/
And getting Coverage of 72.51%

**Review by praveen-garg — APPROVED** [review]
I assume all omitted stuff is a dead code now, given the C Python interface?!

File: tests/unit/python/.coveragerc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    */__template__.py
+    */setup.py
+    */python/fledge/tasks/north/sending_process.py
+    # Omit directory
+    */python/fledge/plugins/common/*
+    */python/fledge/plugins/filter/*
+    */python/fledge/plugins/north/*
+    */python/fledge/plugins/south/*
+    */python/fledge/plugins/notificationDelivery/*
+    */python/fledge/plugins/notificationRule/*
```
Comment (praveen-garg) [inline tests/unit/python/.coveragerc:14]: No such directories exist.

**Comment by praveen-garg** [comment]
> is there any test job run on latest commit this pull request?
> 
> I ran this nightly test: http://jenkins.dianomic.com:8080/view/Fledge%20Nightly%20Jobs/job/fledge_nightly_build_ub2004/1248/ And getting Coverage of 72.51%

@Mohit04tomar Have you used `--cov-config` arg with rc file?

**Comment by ashish-jabble** [comment]
> is there any test job run on latest commit this pull request?

NO, as I couldn't find any custom Job for that; only tested with local enviroment! Other reason to not use nightly build as this will disturb the trend of coverage of daily basis. We should provide a custom Job to tun Coverage along with unit tests

> I ran this nightly test:
http://jenkins.dianomic.com:8080/view/Fledge%20Nightly%20Jobs/job/fledge_nightly_build_ub2004/1248/
And getting Coverage of 72.51%

Groovy script also required a change to run with --cov-config as Praveen has already pointed out in above comment.

File: tests/unit/python/.coveragerc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[run]
+omit =
+    # Ignore files
+    */__init__.py
+    */__template__.py
+    */setup.py
+    */python/fledge/tasks/north/sending_process.py
+    # Omit directory
+    */python/fledge/plugins/common/*
```
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply tests/unit/python/.coveragerc:9]: `python/fledge/plugins/storage` has backup and restore python based module

File: tests/unit/python/.coveragerc (hunk 1 above)
Comment (ashish-jabble) [reply tests/unit/python/.coveragerc:14]: correct, these are only once we installed python based notification plugins. This is only for local setup if anyone wants to check coverage report.

File: tests/unit/python/.coveragerc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[run]
+omit =
+    # Ignore files
+    */__init__.py
+    */__template__.py
+    */setup.py
+    */python/fledge/tasks/north/sending_process.py
```
[the remark this answers is not in the document]
Comment (ashish-jabble) [reply tests/unit/python/.coveragerc:7]: sure

**Review by Mohit04tomar — APPROVED** [review]

**Pull request merged**